processors/zipkin/src/main/java/org/hawkular/apm/processor/zipkin/CommunicationDetailsDeriver.java

```diff
@@ -0,0 +1,172 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.hawkular.apm.processor.zipkin;
+
+import java.net.URL;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import javax.annotation.PostConstruct;
+import javax.inject.Inject;
+
+import org.hawkular.apm.api.model.events.CommunicationDetails;
+import org.hawkular.apm.api.model.events.ProducerInfo;
+import org.hawkular.apm.api.utils.EndpointUtil;
+import org.hawkular.apm.server.api.model.zipkin.Span;
+import org.hawkular.apm.server.api.services.ProducerInfoCache;
+import org.hawkular.apm.server.api.task.AbstractProcessor;
+import org.hawkular.apm.server.api.task.RetryAttemptException;
+import org.hawkular.apm.server.api.utils.ProducerInfoCacheUtil;
+
+/**
+ * This class represents the zipkin communication details deriver.
+ *
+ * @author gbrown
+ */
+public class CommunicationDetailsDeriver extends AbstractProcessor<Span, CommunicationDetails> {
+
+    private static final Logger log = Logger.getLogger(CommunicationDetailsDeriver.class.getName());
+
+    @Inject
+    private ProducerInfoCache producerInfoCache;
+
+    private ProducerInfoCacheUtil producerInfoInitialiser;
+
+    /**
+     * The default constructor.
+     */
+    public CommunicationDetailsDeriver() {
+        super(ProcessorType.OneToOne);
+    }
+
+    @PostConstruct
+    public void init() {
+        producerInfoInitialiser = new ProducerInfoCacheUtil();
+        producerInfoInitialiser.setProducerInfoCache(producerInfoCache);
+    }
+
+    /**
+     * @return the producerInfoCache
+     */
+    public ProducerInfoCache getProducerInfoCache() {
+        return producerInfoCache;
+    }
+
+    /**
+     * @param producerInfoCache the producerInfoCache to set
+     */
+    public void setProducerInfoCache(ProducerInfoCache producerInfoCache) {
+        this.producerInfoCache = producerInfoCache;
+    }
+
+    /* (non-Javadoc)
+     * @see org.hawkular.apm.server.api.task.Processor#initialise(java.util.List)
+     */
+    @Override
+    public void initialise(String tenantId, List<Span> items) throws RetryAttemptException {
+        producerInfoInitialiser.initialiseFromSpans(tenantId, items);
+    }
+
+    /* (non-Javadoc)
+     * @see org.hawkular.apm.server.api.task.Processor#processSingle(java.lang.Object)
+     */
+    @Override
+    public CommunicationDetails processOneToOne(String tenantId, Span item) throws RetryAttemptException {
```

Comment:
It is named after the generic API's parameter (autogenerated by Eclipse). Would prefer to keep the generic name, as consistent across implementations, unless you really think it is a problem?